mesh/profiles/views.py

```diff
@@ -44,3 +60,169 @@ def profile_picture(request):
                 response.update({"status": "error"})
                 response.update({"message": "An account does not exist with this account ID."})
                 return JsonResponse(response)
+            
+class ProfilePicturesView(View):
+    """
+        Handles HTTP requests related to Profile Pictures, supporting
+        POST to upload a profile picture,
+        PATCH to update a profile picture,
+        DELETE to delete a profile picture.
+    """
+    
+    def post(self, request, *args, **kwargs):
+        """
+            Handles POST requests.
+
+            Grabs an existing Profile and uploads a profile picture to imgur,
+            and saves the link with the Profile.
+
+            The required fields are "accountID" and "profilePicture".
+
+            Note: A user should not have a profile picture by this point, as in their
+            profilePicture URL should be empty. If they already have a picture, 
+            a PATCH request should be made instead. If a user wishes to delete their picture,
+            a DELETE request should be made instead.
+        """
+        
+        try:
+            data = request.POST
+
+            account_id = data["accountID"]
+            
+            profile = Profile.objects.get(accountID=account_id)
+            
+            # Check if user already has a profilePicture
+            if profile.profilePicture is not None:
+                return JsonResponse({"error": "Profile already has a profilePicture. " +
+                                     "Use PATCH to update the profilePicture or " + 
+                                     "DELETE to delete the profilePicture."}, 
+                                    status = 409)
```

Comment:
Shouldn't the status code be 405?

Reply:
My thought is that it should be 409 rather than 405, as "The HTTP 409 status code (Conflict) indicates that the request could not be processed because of conflict in the request, such as the requested resource is not in the expected state, or the result of processing the request would create a conflict within the resource."

405 would mean that we recognize the request method, but it isn't allowed, which isn't the case as we do what to allow users to POST or PATCH.